eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/RegenerateUrlAliasesCommand.php

```diff
@@ -0,0 +1,1054 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ *
+ * @version //autogentag//
+ */
+namespace eZ\Bundle\EzPublishMigrationBundle\Command\LegacyStorage;
+
+use eZ\Publish\API\Repository\Exceptions\ForbiddenException;
+use eZ\Publish\Core\Persistence\Legacy\Content\UrlAlias\Gateway as UrlAliasGateway;
+use eZ\Publish\Core\Persistence\Legacy\Content\UrlAlias\Handler as UrlAliasHandler;
+use eZ\Publish\Core\Persistence\Legacy\Handler as LegacyStorageEngine;
+use eZ\Publish\SPI\Persistence\Content\UrlAlias;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\DBAL\Schema\Schema;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\OutputInterface;
+use RuntimeException;
+use Exception;
+use PDO;
+
+class RegenerateUrlAliasesCommand extends ContainerAwareCommand
+{
+    const MIGRATION_TABLE = '__migration_ezurlalias_ml';
+    const CUSTOM_ALIAS_BACKUP_TABLE = '__migration_backup_custom_alias';
+    const GLOBAL_ALIAS_BACKUP_TABLE = '__migration_backup_global_alias';
+
+    /**
+     * @var \eZ\Publish\API\Repository\ContentService
+     */
+    protected $contentService;
+
+    /**
+     * @var \eZ\Publish\Core\Repository\Helper\NameSchemaService
+     */
+    protected $nameSchemaResolver;
+
+    /**
+     * @var \eZ\Publish\SPI\Persistence\Content\UrlAlias\Handler
+     */
+    protected $urlAliasHandler;
+
+    /**
+     * @var \eZ\Publish\Core\Persistence\Legacy\Content\UrlAlias\Gateway
+     */
+    protected $urlAliasGateway;
+
+    /**
+     * @var \Doctrine\DBAL\Connection
+     */
+    protected $connection;
+
+    /**
+     * @var int
+     */
+    protected $bulkCount;
+
+    /**
+     * @var \Symfony\Component\Console\Output\OutputInterface
+     */
+    protected $output;
+
+    protected $actionSet = [
+        'full' => true,
+        'autogenerate' => true,
+        'backup-custom' => true,
+        'restore-custom' => true,
+        'backup-global' => true,
+        'restore-global' => true,
+    ];
+
+    protected function configure()
+    {
+        $this
+            ->setName('ezpublish:regenerate:legacy_storage_url_aliases')
+            ->setDescription('Updates sort keys in configured Legacy Storage database')
```

Comment:
Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/1781/commits/20145c6dba5bcfd6910d996145287d88340398e6